### PR TITLE
Driver factory branch

### PR DESCRIPTION
**Pull request by Artdianic94**
Here I added a multi browsers

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+      - "ENABLE_VNC=true"
+      - "SCREEN_RESOLUTION=1920x1080x24"
+    volumes:
+      - /dev/shm:/dev/shm
+    command: ["--listen", "0.0.0.0:4444", "--selenoid-uri", "http://selenoid:4444/wd/hub"]
+    networks:
+      - network
+    container_name: firefox
+
+  opera:
```
Comment (rmarchuk) [inline docker-compose.yml:51]: there is no need in opera support

File: config/browsers.json [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    "default": "latest",
+    "versions": {
+      "latest": {
+        "image": "selenoid/firefox:latest",
+        "port": "4444",
+        "path": "/"
+      }
+    }
+  },
+  "opera": {
```
Comment (rmarchuk) [inline config/browsers.json:22]: there is no need in opera support

**Review by rmarchuk — CHANGES_REQUESTED** [review]

File: src/test/java/tests/TestBase.java [hunk 3]
```diff
@@ -17,17 +19,41 @@
...
+                driverType = DriverType.CHROME;
+                break;
+            case "firefox":
+                driverType = DriverType.FIREFOX;
+                break;
+            case "remote":
+                driverType = DriverType.REMOTE;
+                break;
+            default:
+                driverType = null;
```
[the remark this answers is not in the document]
Comment (Artdianic94) [reply src/test/java/tests/TestBase.java:41]: fixed

File: src/test/java/tests/TestBase.java [hunk 4]
```diff
@@ -17,17 +19,41 @@
...
+        }
+        driverManager = factory.getManager(driverType);
+        driverManager.setUpDriver();
+        driver = driverManager.getDriver();
+    }
+
+    public void startBrowser() {
```
[the remark this answers is not in the document]
Comment (Artdianic94) [reply src/test/java/tests/TestBase.java:48]: one line method was deleted

File: src/test/java/tests/TestBase.java [hunk 5]
```diff
@@ -17,17 +19,41 @@
...
 
 public class TestBase {
-    public WebDriver driver;
-    public static ChromeDriverManager chromeDriverManager;
+    WebDriver driver;
+    DriverManager driverManager;
     @RegisterExtension
     AfterEachExtension afterEachExtension = new AfterEachExtension();
 
+    public void setUp(String browser) {
```
[the remark this answers is not in the document]
Comment (Artdianic94) [reply src/test/java/tests/TestBase.java:27]: fixed

File: src/test/java/factorydriver/DriverFactory.java [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+                driverManager = new ChromeDriverManager();
+                break;
+            case FIREFOX:
+                driverManager = new FirefoxDriverManager();
+                break;
+            case REMOTE:
+                driverManager = new RemoteDriverManager();
+                break;
+            default:
+                throw new IllegalStateException("Unexpected driver type: " + type);
```
[the remark this answers is not in the document]
Comment (Artdianic94) [reply src/test/java/factorydriver/DriverFactory.java:20]: fixed

File: src/test/java/factorydriver/RemoteDriverManager.java [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+                        "enableVNC", true,
+                        "enableVideo", true
+                ));
+                try {
+                    driver = new RemoteWebDriver(new URL(HUB_URL), capabilities);
+                } catch (MalformedURLException e) {
+                    e.printStackTrace();
+                }
+                break;
+            case "opera":
```
[the remark this answers is not in the document]
Comment (Artdianic94) [reply src/test/java/factorydriver/RemoteDriverManager.java:52]: this case was deleted in another pull request

File: src/test/java/factorydriver/RemoteDriverManager.java [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+        if (remoteBrowser == null) {
+            throw new IllegalArgumentException("Remote browser is not specified!");
+        }
+
+        DesiredCapabilities capabilities = new DesiredCapabilities();
+        capabilities.setCapability("enableVNC", true);
+        capabilities.setCapability("enableVideo", true);
+
+        switch (remoteBrowser.toLowerCase()) {
+            case "chrome":
```
[the remark this answers is not in the document]
Comment (Artdianic94) [reply src/test/java/factorydriver/RemoteDriverManager.java:26]: code duplication was deleted in another pull request

File: docker-compose.yml (hunk 1 above)
Comment (Artdianic94) [reply docker-compose.yml:51]: opera was deleted in another pull request

File: config/browsers.json (hunk 2 above)
Comment (Artdianic94) [reply config/browsers.json:22]: opera was deleted in another pull request

**Review by rmarchuk — APPROVED** [review]

**Pull request merged**